src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/LearnerChangeOfCircumstance/LearnerChangeOfCircumstanceCommandHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.LearnerChangeOfCircumstance
+{
+    public class LearnerChangeOfCircumstanceCommandHandler : ICommandHandler<LearnerChangeOfCircumstanceCommand>
+    {
+        private readonly IApprenticeshipIncentiveDomainRepository _domainRepository;
+        private readonly ILearnerDomainRepository _learnerDomainRepository;
+
+        public LearnerChangeOfCircumstanceCommandHandler(IApprenticeshipIncentiveDomainRepository domainRepository, ILearnerDomainRepository learnerDomainRepository)
+        {
+            _domainRepository = domainRepository;
+            _learnerDomainRepository = learnerDomainRepository;
+        }
+
+        public async Task Handle(LearnerChangeOfCircumstanceCommand command, CancellationToken cancellationToken = default)
+        {
+            var incentive = await _domainRepository.Find(command.ApprenticeshipIncentiveId);
+
+            if (!incentive.HasPossibleChangeOfCircumstances)
+            {
+                return;
+            }
+
+            var learner = await _learnerDomainRepository.GetOrCreate(incentive);
+
+            if (learner.SubmissionFound && learner.SubmissionData.StartDate.HasValue)
```

Comment:
Should this rule be in the domain object along with the HasPossibleChangeOfCircumstances logic?

So have something like 

incentive.SetChangeOfCircumstances(ILearnerDomainRepository  learnerRepository)

?  

Then all the business rule is in the domain object rather than the handler?